File: 1855-maximum-distance-between-a-pair-of-values/1855-maximum-distance-between-a-pair-of-values.py
class Solution:
    def maxDistance(self, A: List[int], B: List[int]) -> int:
        i, j = 0, 1

        while i < len(A) and j < len(B):
            i += A[i] > B[j]
            j += 1

        return j - i - 1

# array, two pointers
# time: O(n)
# space: O(1)

File: 1855-maximum-distance-between-a-pair-of-values/1855-maximum-distance-between-a-pair-of-values.go
func maxDistance(nums1 []int, nums2 []int) int {
    ans := 0
    for i := 0; i < len(nums1); i++ {
        for j := i + ans; j < len(nums2); j++ {
            if nums1[i] > nums2[j] {
                break
            }

            ans = max(ans, j - i)
        }
    }

    return ans
}

// array
// time: O(nm)
// space: O(1)